chess-AI/stockfish_train.py

```diff
@@ -0,0 +1,109 @@
+import random
+import numpy as np
+from stockfish import Stockfish
+import chess
+
+from output_representation import PlayNetworkPolicyConverter
+
+
+class StockfishTrain:
+    """Class with all the necessary functionialities for
+    building a dataset using stockfish
+
+    Attributes:
+        stockfish: the stockfish object
+        policy_converter: reference to the Policy Converter Class
+    """
+
+    def __init__(self, path):
+        self.stockfish = Stockfish(path)
+        self.policy_converter = PlayNetworkPolicyConverter()
+
+    def set_params(self, dashboard):
+        """ Sets the elo and depth for stockfish using the dashboard
+
+        Parameters:
+        dashboard: reference to the streamlit dashboard (this is temporary)
+        """
+
+        elo, depth = dashboard.configure_stockfish()
+        
+        # Set the elo and depth of the stockfish object
+        self.stockfish.set_elo_rating(elo)
+        self.stockfish.set_depth(depth)
+
+    def sig(self, value, scale):
+        """Calculate the sigmoid of a value
+
+        Parameters:
+        value: The value to take the sigmoid of
+        scale: How much to scale the value by
+        """
+
+        return 1 / (1 + np.exp(value/scale))
+
+    def choose_move(self, moves):
+        """ Chooses the next move to make
+        We want to make sure the top move isn't always what is
+        chosen
```

Comment:
```py
"""Choose the next move to make with an epsilon-greedy strategy.

To encourage exploration, a random move is chosen with probability 1-epsilon.
Otherwise, the best action (according to the stockfish evaluation) is chosen.
"""